### ISSUE TITLE
$transclude is not available in hybrid Angular/AngularJS apps

### ISSUE THREAD

**Issue by nfriend** [angular/angular#23268]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
When using an AngularJS component in a hybrid Angular/AngularJS application, the `$transclude` injectable is not available to the AngularJS component's controller.

When an AngularJS component controller asks for `$transclude`, this error is thrown in the browser's console:

```
Error: [$injector:unpr] Unknown provider: $transcludeProvider <- $transclude
http://errors.angularjs.org/1.6.9/$injector/unpr?p0=%24transcludeProvider%20%3C-%20%24transclude
    at eval (VM2834 angular.js:116)
    at eval (VM2834 angular.js:4887)
    at Object.getService [as get] (VM2834 angular.js:5047)
    at eval (VM2834 angular.js:4892)
    at getService (VM2834 angular.js:5047)
    at injectionArgs (VM2834 angular.js:5072)
    at Object.instantiate (VM2834 angular.js:5118)
    at UpgradeHelper.$controller (VM2834 angular.js:11091)
    at UpgradeHelper.buildController (VM2837 upgrade-static.umd.js:867)
    at Ng1DemoComponentFacade.UpgradeComponent.ngOnInit (VM2837 upgrade-static.umd.js:1132)
```

## Expected behavior
The `$transclude` service/local should be injected into the AngularJS component's controller as it would when running inside a pure AngularJS app.

## Minimal reproduction of the problem with instructions

Minimal demo: https://stackblitz.com/edit/transclude-hybrid-error?file=ng1%2Fng1-demo.component.ts

Open up the browser console to see the error that is thrown.

## What is the motivation / use case for changing the behavior?

AngularJS components that use `$transclude` should be able to be used in an Angular app.

Or, if this isn't a supported use case, a note should be added to the documentation that this AngularJS feature isn't supported in hybrid apps.

## Environment
<pre><code>Angular version: 5.2.4
AngularJS version: 1.6.9

Browser: any
</code></pre>

**Comment by nfriend** [comment]
This issue was mentioned as a side note in https://github.com/angular/angular/issues/22175:

> Aside of that, it appears that `$transclude` local is not available either and thus cannot be injected ?

but it appears to not have been addressed as part of the fix.

**Comment by gkalpak** [comment]
Duplicate of #22824. See #22824 and #22832 for discussion.

**Comment by FDIM** [comment]
@nfriend Could you tell a bit more about your use case? If you are compiling the template dynamically (changing context/scope etc) then it's simply not possible (difficult / not worth the effort?) to make it work. On the other hand, if you are only checking for used slots or simply transcluding the content as is then the PR that is referenced could be finished to support that.

**Comment by nfriend** [comment]
@FDIM - Thanks for following up. In my case, I'm only using the `$transclude` local to test if a transclusion slot is filled, i.e:

```TypeScript
$transclude.isSlotFilled('myTransclusionSlot');
```

**Comment by FDIM** [comment]
Okay so you have a similar use case and such support is actually possible and done in the PR #22832 . It's up to @gkalpak to decide if that PR should go in (of course after I make few fixes to it based on comments). I don't mind much as we are blocked by other issue and using local copy of the library anyway.
